### PR TITLE
resolved conflict with local branch, Updated ConnectionView where you…

### PR DESCRIPTION
**Pull request by ghost**
… can choose to connect to different clients, not error safe, BoardViewUpdater fix

### CODE REVIEW

File: server/src/main/java/com/blackship/battlesheep/game/state/GameStateInProgress.java [hunk 1]
```diff
@@ -30,8 +30,8 @@ public GameState changeState(List<List<Integer>> firstPlayerPositions, List<List
         firstPlayerFleet.shootShipPositions(firstPlayerMoves);
         secondPlayerFleet.shootShipPositions(secondPlayerMoves);
 
-        if(firstPlayerFleet.isSunk()) throw new FirstPlayerWon();
-        if(secondPlayerFleet.isSunk()) throw new SecondPlayerWon();
+        //if(firstPlayerFleet.isSunk()) throw new FirstPlayerWon();
+        //if(secondPlayerFleet.isSunk()) throw new SecondPlayerWon();
```
Comment (Dzionek95) [inline server/src/main/java/com/blackship/battlesheep/game/state/GameStateInProgress.java:35]: comments?

**Review by Dzionek95 — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java [hunk 2]
```diff
@@ -1,10 +1,11 @@
 package com.blackship.battlesheep.fx;
 
-import com.blackship.battlesheep.fx.controllers.SalvoHandler;
 import com.blackship.battlesheep.fx.utils.ButtonUtils;
 import javafx.scene.control.Button;
+import javafx.scene.control.ButtonBuilder;
 
 import java.util.*;
+import java.util.stream.Collectors;
```
Comment (Devas) [inline client/src/main/java/com/blackship/battlesheep/fx/BoardViewUpdater.java:8]: :+1:

File: client/src/main/java/com/blackship/battlesheep/fx/ClientCommunicationHandlerKeeper.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.blackship.battlesheep.fx;
+
+import com.blackship.battlesheep.AppClient;
+import com.blackship.battlesheep.communication.network.AppClientCommunicationHandler;
+import com.blackship.battlesheep.communication.network.AppClientSocket;
+
+/**
+ * Created by Mateusz on 04.08.2017.
+ */
+public enum ClientCommunicationHandlerKeeper {
```
Comment (Devas) [inline client/src/main/java/com/blackship/battlesheep/fx/ClientCommunicationHandlerKeeper.java:10]: Singleton? Can you refactor it?

File: server/src/main/java/com/blackship/battlesheep/game/state/GameStateInProgress.java [hunk 4]
```diff
@@ -30,8 +30,8 @@ public GameState changeState(List<List<Integer>> firstPlayerPositions, List<List
         firstPlayerFleet.shootShipPositions(firstPlayerMoves);
         secondPlayerFleet.shootShipPositions(secondPlayerMoves);
 
-        if(firstPlayerFleet.isSunk()) throw new FirstPlayerWon();
-        if(secondPlayerFleet.isSunk()) throw new SecondPlayerWon();
+        //if(firstPlayerFleet.isSunk()) throw new FirstPlayerWon();
```
Comment (Devas) [inline server/src/main/java/com/blackship/battlesheep/game/state/GameStateInProgress.java:33]: Production should not have comments with code

**Review by Devas — APPROVED** [review]

**Pull request merged**